### ISSUE TITLE
Liquidity Management

### ISSUE THREAD

**Issue by kjr217** [rysk-finance/dynamic-hedging#7]
In a pool, when writing options there are different collateral requirements depending on the option being written. So writing a put option warrants USD collateral whereas writing a call option warrants ETH/BTC/X collateral. Therefore, a pool must have both ETH and USD present. There are a number of possible solutions, this thread aims to explore possible options for this issue.

1. Depositing both ETH and USD into the pool. 
 - Problem: this has the effect of influencing the delta of the pool as ETH will produce a positive pool delta. (Although this could work in favour if this is used as delta management strategy)
2. Only depositing USD into the pool
- Problem: this will mean that you will need to swap into ETH which will incur slippage costs and swap fees.
3. Balancer pool mechanism: @thegeronimo can provide more details

Please provide more options :)

**Comment by thegeronimo** [comment]
Regarding having both USD and ETH(or other underlying) in the pool it would be the simplest approach and would initially put the pool delta at 0.5. The pool would then naturally hedge this out by quoting options against this. 

The question then becomes do we want to manage non-options asset allocation? If we do then we can keep liquidity in a balancer pool and use exposure control (https://docs.balancer.fi/products/balancer-pools/weighted-pools#exposure-control) to maintain a target asset allocation (50%/50% or perhaps dynamically adjusting the weights as needed).

**Comment by kjr217** [comment]
Right, but if we then allow people to put any amount of ETH in they can force cheaper options prices if they pump delta in a specific direction. I guess doing 50 50 works if we use some other way of immediately hedging the exposure (as i think issuing options immediately to hedge could be problematic)

I think we should do more research into dynamically weighting a balancer pool

**Comment by thegeronimo** [comment]
It would be helpful to have a specific attack case with state transitions. If they put ETH in the delta would shift up and they could then buy calls cheaper but then instantly redeeming those shares at the same value would be challenging for them.

**Comment by kjr217** [comment]
Pool delta is at 0.

User A takes out a flash loan for 1000 eth and deposits 1000 eth into the pool (with 1 eth in their account), this pushes delta to +1000. The user then purchases the call at a cheaper cost because purchasing the call will reduce portfolio delta, they can then withdraw their 1000 eth and repay the flash loan, leaving with their cheaper call. Then they can sell this call back to the pool and make an instant profit as portfolio delta is back to normal so calls are priced more reasonably.

**Comment by thegeronimo** [comment]
There are a few missing variables here. What is the initial state of the pool liquidity? How many calls do they buy? 

The pool minting calls will lockup ETH as collateral so withdrawing that same ETH at that moment would not be possible. Unless we change the spec the pool would not be a buyer of those calls at that moment because it would push delta further away from the zero target, but if it was a buyer the price would be at a discount to what the holder paid for them. Even in the case where it pushes delta towards target there will be a spread between selling price and buying price for the same options because selling options use an IV that is expected to be greater than realized volatility and buying will use IV that is expected to be less than realized volatility. 

Once the pool writes those calls their fair value is marked as liabilities on the balance sheet of the pool. The value of shares are `assets - liabilities`  so even if the pool has liquidity available for share redemption, it's worth considering what the value of collateral locked in the calls is, as it's no longer just liquid collateral, it's locked collateral at risk in addition to the liability side increasing so the share value would likely not be higher at that moment.

**Comment by kjr217** [comment]
Picked up on a call: 

- Need to refine how the spread works to make sure this is an avoidable attack vector
